compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/HeaderInfo.kt

```diff
@@ -106,12 +71,18 @@ internal class ProgressionHeaderInfo(
     additionalNotEmptyCondition = additionalNotEmptyCondition
 ) {
 
+    @ExperimentalUnsignedTypes
```

Comment:
I feel like annotating all files which use unsigned API with `@file:OptIn(ExperimentalUnsingedTypes::class)` would be a bit less intrusive

Reply:
I do prefer it that way too. Although... I don't know if it's an IDE bug, but if I use `@file:OptIn(...)` in `ProgressionType.kt`, IntelliJ gives still gives me a warning on the usages in `UIntProgressionType` and `ULongProgressionType` on `override fun DeclarationIrBuilder.minValueExpression() = irInt(UInt.MIN_VALUE.toInt())`. Very weird because I use `UInt.MIN_VALUE` in the super-constructor call. That is the reason I annotated the members instead. Anyway I changed it to use `@file`, we can choose to ignore this for now.

Reply:
IDE issue looks like https://youtrack.jetbrains.com/issue/KT-30264